opencti-platform/opencti-front/src/private/components/data/connectors/ConnectorWorks.tsx

```diff
@@ -0,0 +1,359 @@
+import { useFormatter } from '../../../../components/i18n';
+import { FIVE_SECONDS } from '../../../../utils/Time';
+import { commitMutation, MESSAGING$ } from '../../../../relay/environment';
+import Transition from '../../../../components/Transition';
+
+const interval$ = interval(FIVE_SECONDS);
+
+const useStyles = makeStyles(() => ({
+  paper: {
+    height: '100%',
+    minHeight: '100%',
+    margin: '10px 0 20px 0',
+    padding: '15px',
+    borderRadius: 6,
+    position: 'relative',
+  },
+  number: {
+    fontWeight: 600,
+    fontSize: 18,
+  },
+  progress: {
+    borderRadius: 5,
+    height: 10,
+  },
+}));
+
+export const connectorWorksWorkDeletionMutation = graphql`
+  mutation ConnectorWorksWorkDeletionMutation($id: ID!) {
+    workEdit(id: $id) {
+      delete
+    }
+  }
+`;
+
+type WorkMessages = NonNullable<NonNullable<NonNullable<ConnectorWorks_data$data['works']>['edges']>[0]>['node']['errors'];
+
+interface Options {
+  count: number,
+  filters: {
+    mode: string,
+    filters: [
+      {
+        key: string
+        values: string[]
+        operator: string
+        mode: string
+      },
+    ],
+    filterGroups: [],
+  },
+}
+
+interface ConnectorWorksComponentProps {
+  data: ConnectorWorks_data$data
+  options: Options[]
+  relay: RelayRefetchProp
+}
+
+const ConnectorWorksComponent: FunctionComponent<ConnectorWorksComponentProps> = ({ data, options, relay }) => {
+  const works = data.works?.edges ?? [];
+  const { t_i18n, nsdt } = useFormatter();
+  const classes = useStyles();
+  const [displayErrors, setDisplayErrors] = useState<boolean>(false);
+  const [errors, setErrors] = useState<WorkMessages>([]);
+
+  const handleOpenErrors = (errorsList: WorkMessages) => {
+    if (!errorsList) return;
+    setDisplayErrors(true);
+    setErrors(errorsList);
+  };
+
+  const handleCloseErrors = () => {
+    setDisplayErrors(false);
+    setErrors([]);
+  };
+
+  // eslint-disable-next-line class-methods-use-this
+  const handleDeleteWork = (workId: string) => {
+    commitMutation({
+      mutation: connectorWorksWorkDeletionMutation,
+      variables: {
+        id: workId,
+      },
+      updater: undefined,
+      optimisticUpdater: undefined,
+      optimisticResponse: undefined,
+      onCompleted: () => {
+        MESSAGING$.notifySuccess('The work has been deleted');
+      },
+      onError: undefined,
+      setSubmitting: undefined,
+    });
+  };
+
+  useEffect(() => {
+    const subscription = interval$.subscribe(() => {
+      relay.refetch(options);
+    });
+    return () => subscription.unsubscribe();
+  }, []);
+
+  return (
+    <div>
+      {works.length === 0 && (
+        <Paper
+          classes={{ root: classes.paper }}
+          variant="outlined"
+        >
+          <Typography align='center'>
+            {t_i18n('No work')}
+          </Typography>
+        </Paper>
+      )}
+      {works.map((workEdge) => {
+        const work = workEdge?.node;
+        if (!work) return null;
```

Comment:
I think you should check the graphQL, I don't see why we could have null workEdge here

Reply:
`works.edges` can be `null` or Array of (`object` or `null`), because the current relay generated type is :
```
export type ConnectorWorks_data$data = {
  readonly works: {
    readonly edges: ReadonlyArray<{...} | null> | null;
  } | null;
};
```

Reply:
GraphQL types :
```
type WorkConnection {
  pageInfo: PageInfo!
  edges: [WorkEdge]               <= edges can be null / WorkEdge can be null
}

type WorkEdge {
  cursor: String!
  node: Work!                          <= node can be null
}

type Work {
  id: ID!
  name: String
  user: Creator
  connector: Connector
  timestamp: DateTime!
  status: State!
  event_source_id: String
  received_time: DateTime
  processed_time: DateTime
  completed_time: DateTime
  completed_number: Int
  messages: [WorkMessage]
  errors: [WorkMessage]
  tracking: WorkTracking
}
```

Reply:
@Kedae I had the same remark initially, but it appears that the gql types are defined with possible null at all levels, even inside arrays... It seems strange, but changing this requires checking the actual implementation backend side, see if it's _really_ not possible.